### PR TITLE
[Merged by Bors] - feat(Fintype.Basic): Add two theorems supporting Auction Theory formalizations

### PR DESCRIPTION
**Pull request by hcWang942**
This PR adds two missing theorems in `Mathlib.Data.Fintype.Basic`

---

This is used in another PR of mine formalizing some basic theorems in `GameTheory.AuctionTheory.Basic` .

One may check the previous PR for Auction Theory here: #13248 

I would sincerely appreciate a timely review of this PR, as I am eager to complete my first project on Auction Theory before my upcoming birthday. Thank you very much for your consideration and support!

### CODE REVIEW

**Comment by Shreyas4991** [comment]
It seems you have rebased your PR on the other PR branch so now we have all these other commits. Can you remove those commits, branch off master (or reset HEAD to the latest master commit) and then commit only the two new theorems?

**Comment by hcWang942** [comment]
> It seems you have rebased your PR on the other PR branch so now we have all these other commits which are not needed in this PR. In your branch `finset_nontrivial_two_theorems_add` can you remove those commits by resetting `HEAD` to the latest master commit and then commit only the two new theorems? Once you only have the one new commit in this branch for adding the two theorems, you will have to force push to this branch here.

May I know if it is correct to use the command
`
git fetch origin master
git reset --hard origin/master
`
Then commit and push the changes again.

File: Mathlib/Data/Fintype/Basic.lean
Comment (tb65536) [inline Mathlib/Data/Fintype/Basic.lean:117]: I think the `[Fintype α]` assumption is already present at this point of the file.

**Comment by Shreyas4991** [comment]
> You don't need to `git fetch origin master`. I have a beginner friendly but slightly tedious idea. I am assuming you are currently on this branch at the last I commit I see on this page [9659d19](https://github.com/leanprover-community/mathlib4/pull/14163/commits/9659d19481b1b158d7fde7bc9d79bae15e624b07). if you open git log by running `git log` and scroll down, you will find the most recent commit that is not your commit in this PR. copy that commit ID (which I denote by `<commit_ID>` hereon). then `git reset --hard <commit_ID>`. Then add your two theorems (which you can in the respective file). add and commit that. Then `git push --force origin finset_nontrivial_two_theorems_add`

I am so so sorry. Instead of hitting reply, I seem to have hit edit on your comment. I hope the question you have was answered though.

**Comment by Shreyas4991** [comment]
Could you please delete this file `https://github.com/leanprover-community/mathlib4/pull/14163/files#diff-ffd24c8f58ab0af7a8d4610ab7fc208cf0fef60e76059dcd605f7113179a96ca`

**Comment by hcWang942** [comment]
> Could you please delete this file `https://github.com/leanprover-community/mathlib4/pull/14163/files#diff-ffd24c8f58ab0af7a8d4610ab7fc208cf0fef60e76059dcd605f7113179a96ca`
> 
> EDIT: You can do this from this webpage by going to the `files changed` tab at the top your PR description and deleting that large file with 66000 changes.
> 
> EDIT 2 : If I have to guess the cause of this file, git probably opened your text editor for the commit message after the reset and for reasons beyond me saved the file into the toplevel directory of your local clone of mathlib. This file can be safely discarded.

Thanks for the suggestion, relevant modification has been done!

**Review by Shreyas4991 — COMMENTED** [review]
I am not convinced that `univ_nontrivial` belongs in mathlib. I don't think it is mathlib practice to have `mp` and  `mpr` versions of lemmas as separate globally named lemmas. The `iff` version suffices. `univ_nontrivial_iff` can be used with `rw` and `univ_nontrivial_iff.mpr` with `apply`, depending on which tactic you are using.

**Comment by tb65536** [comment]
The one advantage of `univ_nontrivial` is that it can take `Nontrivial α` automatically. For instance, see `univ_nonempty_iff` and `univ_nonempty` a few lines above.

**Comment by Shreyas4991** [comment]
> The one advantage of `univ_nontrivial` is that it can take `Nontrivial α` automatically. For instance, see `univ_nonempty_iff` and `univ_nonempty` a few lines above.

If so, is there a reason this theorem is not an instance declaration instead?

**Comment by tb65536** [comment]
> > The one advantage of `univ_nontrivial` is that it can take `Nontrivial α` automatically. For instance, see `univ_nonempty_iff` and `univ_nonempty` a few lines above.
> 
> If so, is there a reason this theorem is not an instance declaration instead?

I think `Finset.Nonempty` and `Finset.Nontrivial` are just defs, not typeclasses.

**Comment by Shreyas4991** [comment]
> > > The one advantage of `univ_nontrivial` is that it can take `Nontrivial α` automatically. For instance, see `univ_nonempty_iff` and `univ_nonempty` a few lines above.
> > 
> > If so, is there a reason this theorem is not an instance declaration instead?
> 
> I think `Finset.Nonempty` and `Finset.Nontrivial` are just defs, not typeclasses.

Okay then I take back that comment, but I still don't see why `univ_nonempty_iff.mpr ` is not giving you the same ease of use as `univ_nonempty`. That being said, I am not an official reviewer or maintainer, so please take this opinion with a pinch of salt.

**Review by tb65536 — APPROVED** [review]
I think the above discussion has resolved and this is still waiting for review? Personally, this PR looks good to me.

**Comment by hcWang942** [comment]
> I think the above discussion has resolved and this is still waiting for review? Personally, this PR looks good to me.

Thanks for the kind reply, may I know what more can I do now? Or I just need to wait patiently.

**Review by Shreyas4991 — APPROVED** [review]

**Comment by j-loreaux** [comment]
Arguably, `Finset.univ_nontrivial` could / should be marked `@[simp]`, but I think I prefer it as is so that we don't insert this generic typeclass search into the `simp` calls.

bors merge

**Pull request closed without merging**
